# EJERCICIO 1 EXAMEN PRIMER PARCIAL

Se desea estimar el efecto del ingreso (X) sobre el consumo (Y).

In [5]:
#Librerias
import pandas as pd
import numpy as np

In [6]:
#Datos
X = np.array([6,8,10,12,14,16]) #Ingreso
y = np.array([5.1,5.7,6.4,6.9,7.7,8.0]) #Consumo

In [18]:
#MODELO DE REGRESION LINEAL POR MCO

def regresion_lineal(X,y):
    y_media = np.mean(y)
    x_media = np.mean(X)

    b1 = np.sum((X - x_media) * (y - y_media)) / np.sum((X - x_media)**2)
    b0 = y_media - b1*x_media

    print(f"La formula de regresion lineal es: {b0:.2f} + {b1:.2f}X")

    return b0, b1


In [19]:
b0, b1 = regresion_lineal(X,y)

La formula de regresion lineal es: 3.33 + 0.30X


La formula de regresion lineal es: 3.33 + 0.30X, B1 indica que Por cada unidad adicional de ingreso, el consumo aumenta en promedio 0.3014 unidades., b0 = 3.3333 indica que Cuando el ingreso es 0, el consumo estimado sería 3.3333

In [20]:
#Calcula el consumo estimado para X=11.
consumo = b0 + b1*11
print(f"El consumo será de {consumo:.4f} cuando el ingreso sea 11")

El consumo será de 6.6333 cuando el ingreso sea 11


In [21]:
#Calcula el residuo para el hogar con X=10.
y_real = y[X == 10][0]
y_hat = b0 + b1*10
residuo = y_real - y_hat

print(f"El residuo es {residuo:.4f} cuando X vale 10")


El residuo es 0.0667 cuando X vale 10
